src/batch/src/executor/hash_agg.rs

```diff
@@ -212,18 +239,265 @@ impl<K: HashKey + Send + Sync> Executor for HashAggExecutor<K> {
     }
 }
 
+#[derive(Default, Clone, Copy)]
+pub struct SpillBuildHasher(u64);
+
+impl BuildHasher for SpillBuildHasher {
+    type Hasher = XxHash64;
+
+    fn build_hasher(&self) -> Self::Hasher {
+        XxHash64::with_seed(self.0)
+    }
+}
+
+const DEFAULT_SPILL_CHUNK_SIZE: usize = 1024;
+
+/// `AggSpillManager` is used to manage how to write spill data file and read them back.
+/// The spill data first need to be partitioned. Each partition contains 2 files: `agg_state_file` and `input_chunks_file`.
+/// The spill file consume a data chunk and serialize the chunk into a protobuf bytes.
+/// Finally, spill file content will look like the below.
+/// The file write pattern is append-only and the read pattern is sequential scan.
+/// This can maximize the disk IO performance.
+///
+/// ```text
+/// [proto_len]
+/// [proto_bytes]
+/// ...
+/// [proto_len]
+/// [proto_bytes]
+/// ```
+pub struct AggSpillManager {
+    op: SpillOp,
+    partition_num: usize,
+    agg_state_writers: Vec<opendal::Writer>,
+    agg_state_readers: Vec<opendal::Reader>,
+    agg_state_chunk_builder: Vec<DataChunkBuilder>,
+    input_writers: Vec<opendal::Writer>,
```

Comment:
A question: As mentioned in opendal's documents:

> Writer also supports to write with append enabled. This is useful when users want to append some data to the end of the file.
> - If file doesn’t exist, it will be created and just like calling write.
> - If file exists, data will be appended to the end of the file.
> https://opendal.apache.org/docs/rust/opendal/struct.Writer.html#write-with-append-enabled

So I suppose this won't work with AWS S3? Because it doesn't support `append`

Reply:
Every time spill use write to a new file, so I think we can remove the append if we need to spill to S3.

Reply:
Writing a new file for every chunk might be too expensive. Anyway, we may consider this later.